### ISSUE TITLE
Rule request: Coalescing a non-optional

### ISSUE THREAD

**Issue by brennantaylor** [realm/SwiftLint#780]
I don't know what kind of type information if any at all is available to swiftlint but would it be possible to implement this? It's a shame the compiler doesn't complain about it.

```swift
func foo() -> String {
  return "whatever"
}

let a = foo() ?? "never here"
```

**Comment by norio-nomura** [comment]
Swift 3 does:
```console
➜  16:05:14  swiftc --version
Apple Swift version 3.0 (swiftlang-800.0.43.6 clang-800.0.38)
Target: x86_64-apple-macosx10.9
➜  16:06:46  echo '
func foo() -> String {
  return "whatever"
}

let a = foo() ?? "never here"'|swiftc -
<stdin>:6:15: warning: left side of nil coalescing operator '??' has non-optional type 'String', so the right side is never used
let a = foo() ?? "never here"
        ~~~~~ ^~~~~~~~~~~~~~~
              
➜  16:06:51
```

**Comment by brennantaylor** [comment]
Oh nice! Thanks!

I'll close this issue then.
